### PR TITLE
OpenTelemetry Introduction: Confusion is not efficient

### PR DESCRIPTION
**Pull request by alanwest**
The removed section creates more questions than it answers as highlighted in this thread from CNCF Slack https://cloud-native.slack.com/archives/C024DRQ63UP/p1639683781017600.

### CODE REVIEW

**Review by jack-berg — APPROVED** [review]

**Comment by rhetoric101** [comment]
@alanwest, should the tech pubs team wait on this PR until it gets a developer review?

**Comment by alanwest** [comment]
> @alanwest, should the tech pubs team wait on this PR until it gets a developer review?

What's a developer review? Is that part of the normal process?

**Comment by rhetoric101** [comment]
@alanwest, I just saw Jack Berg as a reviewer on this, so I assumed we needed a sign-off from him. If not, I can get this out the door!

**Review by rhetoric101 — APPROVED** [review]
@alanwest, thanks for following up on this Slack comment!

**Comment by alanwest** [comment]
> @alanwest, I just saw Jack Berg as a reviewer on this, so I assumed we needed a sign-off from him. If not, I can get this out the door!

Oh yea no worries, I just wasn't sure if there was a new process. I didn't see Jack had approved it either 😆. He and I were chatting about it.

**Pull request merged**
